kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/StrimziTestClient.java

```diff
@@ -103,20 +103,27 @@ private String waitForConsumer(String namespace, String podName, Duration timeou
         return kubeClient().logsInSpecificNamespace(namespace, podName);
     }
 
-    private List<ConsumerRecord> getConsumerRecords(List<String> logRecords) {
-        return logRecords.stream().map(x -> ConsumerRecord.parseFromJsonString(VALUE_TYPE_REF, x))
-                .filter(Objects::nonNull).map(ConsumerRecord.class::cast).toList();
+    private List<ConsumerRecord> getConsumerRecords(Stream<String> logRecords) {
+        return logRecords.filter(Predicate.not(String::isBlank))
+                .map(x -> ConsumerRecord.parseFromJsonString(VALUE_TYPE_REF, x))
+                .filter(Objects::nonNull)
+                .map(ConsumerRecord.class::cast)
+                .toList();
     }
 
-    private List<String> extractRecordLinesFromLog(String log) {
-        List<String> records = new ArrayList<>();
+    private Stream<String> extractRecordLinesFromLog(String log) {
         String stringToSeek = "Received message:";
 
-        List<String> receivedMessages = Stream.of(log.split("\n")).filter(l -> l.contains(stringToSeek)).toList();
-        for (String receivedMessage : receivedMessages) {
-            records.add(receivedMessage.split(stringToSeek)[1].trim());
-        }
-
-        return records;
+        return Stream.of(log.split("\n"))
+                .filter(l -> l.contains(stringToSeek))
+                .map(line -> {
+                    final String[] split = line.split(stringToSeek, 1);
```

Comment:
```suggestion
                    final String[] split = line.split(stringToSeek);
```
This make the tests work locally in my case

Reply:
Hmm but that implies `stringToSeek` appears multiple times which seems odd but ok.

Reply:
I know, but now it is reporting an `IndexOutOfBounds` don't know why because reading the `limit` parameters it should imply only to look for the first appearance of the string, but it fails.

Reply:
Ahh doh! Your right, I didn't read the limit docs closely enough! I guess a limit of 1 means give me at most 1 elements back where as two means give me at most 2 elements back (thus `limit -1` uses of the regex).